scripts/retrieve_databundle_light.py

```diff
@@ -481,39 +483,32 @@ def get_best_bundles_by_category(
     returned_bundles : list
         List of bundles to download
     """
-    # dictionary with the number of match by configuration for tutorial/non-tutorial configurations
-    dict_n_matched = {
-        bname: config_bundles[bname]["n_matched"]
-        for bname in config_bundles
-        if config_bundles[bname]["category"] == category
-        and config_bundles[bname].get("tutorial", False) == tutorial
-        and _check_disabled_by_opt(config_bundles[bname], config_enable) != ["all"]
-    }
+    df_empty = pd.DataFrame(columns=["bundle_name", "bundle_size", "n_matches"])
+    i = 0
+    for bname in config_bundles:
+        if (
+            config_bundles[bname]["category"] == category
+            and config_bundles[bname].get("tutorial", False) == tutorial
+            and _check_disabled_by_opt(config_bundles[bname], config_enable) != ["all"]
+        ):
+            bundles = pd.DataFrame(
+                {"bundle_name": bname, "n_matches": config_bundles[bname]["n_matched"]},
+                index=[i],
+            )
+            i = i + 1
+            df_empty = pd.concat([df_empty, bundles], axis=0)
```

Comment:
Great Draft @Emre-Yorat89 ! :D
Could you try if this code works? It should a slight improvement of yours

```suggestion
    df_matches = pd.DataFrame(columns=["bundle_name", "bundle_size", "n_matched"])
    for bname, bvalue in config_bundles.items():
        if (
            bvalue["category"] == category
            and bvalue.get("tutorial", False) == tutorial
            and _check_disabled_by_opt(bvalue, config_enable) != ["all"]
        ):
            df_matches.loc[bname] = [bname, len(bvalue["countries"]), bvalue["n_matches"]]
```

Reply:
I have tried. They are working, thank you for the contribution :)